lib/ProductOpener/Products.pm

```diff
@@ -1501,12 +1501,25 @@ sub store_product ($user_id, $product_ref, $comment) {
 
 	$log->debug("store_product - done", {code => $code, product_id => $product_id}) if $log->is_debug();
 
-	my $update_type = $product_ref->{deleted} ? "deleted" : "updated";
+	my $action = "updated";
+
+	if (exists $product_ref->{obsolete}
+		and $product_ref->{obsolete} eq "on")
+	{
+		$action = "archived";
```

Comment:
It's a bit tricky: "archived" products can be "unarchived", and they can also be "updated" (without the obsolete parameter necessarily changing).

I think we should trigger the "archived" event only when the obsolete field is changed to "on", and that we should add an "unarchived" event if it's removed.